notebook/analysis/benchmark_results.py

```diff
@@ -688,7 +693,266 @@ def return_unlabeled():
 
 # %%
 plot = ggplot(tb_metrics, aes(x='train_count', y='balanced_accuracy', color='supervised')) 
-plot += geom_smooth()
+plot += geom_smooth(method='loess')
 plot += geom_point(alpha=.2)
+plot += geom_hline(yintercept=.5, linetype='dashed')
 plot += ggtitle('TB (lack of a) Crossover Point')
 plot
+
+# %% [markdown]
+# ## Small training sets without be correction
+
+# %%
+in_files = glob.glob('../../results/small_subsets.sepsis*.tsv')
+in_files = [file for file in in_files if 'be_corrected' not in file]
+print(in_files[:5])
+
+# %%
+sepsis_metrics = pd.DataFrame()
+for path in in_files:
+    new_df = pd.read_csv(path, sep='\t')
+    model_info = path.strip('.tsv').split('sepsis.')[-1]
+    model_info = model_info.split('.')
+        
+    supervised_model = model_info[0]
+             
+    new_df['supervised'] = supervised_model
+    
+    new_df['seed'] = model_info[-2]
+        
+    sepsis_metrics = pd.concat([sepsis_metrics, new_df])
+    
+sepsis_metrics['train_count'] = sepsis_metrics['train sample count']
+
+# Looking at the training curves, deep_net isn't actually training
+# I need to fix it going forward, but for now I can clean up the visualizations by removing it
+sepsis_metrics = sepsis_metrics[~(sepsis_metrics['supervised'] == 'deep_net')]
+sepsis_metrics['supervised'] = sepsis_metrics['supervised'].str.replace('pytorch_supervised', 'three_layer_net')
+sepsis_metrics
+
+# %%
+plot = ggplot(sepsis_metrics, aes(x='factor(train_count)', y='balanced_accuracy')) 
+plot += geom_boxplot()
+plot += ggtitle('Sepsis Dataset Size Effects (equal label counts)')
+print(plot)
+
+# %%
+plot = ggplot(sepsis_metrics, aes(x='factor(train_count)', y='balanced_accuracy', fill='supervised')) 
+plot += geom_boxplot()
+plot += ggtitle('Sepsis Datset Size by Model (equal label counts)')
+print(plot)
+
+# %%
+plot = ggplot(sepsis_metrics, aes(x='train_count', y='balanced_accuracy', color='supervised')) 
+plot += geom_smooth()
+plot += geom_point(alpha=.2)
+plot += geom_hline(yintercept=.5, linetype='dashed')
+plot += ggtitle('Sepsis Crossover Point')
+plot
```

Comment:
This is interesting! It looks like performance overall is much better without batch correction, and there's no crossover point anymore (logistic regression seems to always do better).

Maybe this could be similar to what Alex's work showed about compendia where many studies were combined? Although you're seeing improved performance even for very small training sets (probably only a few studies), so that doesn't seem like it would explain the whole difference.

Are you just using `limma` for batch effect correction? I wonder if you'd see the same performance difference if you used another method, or if this is particular to `limma`.

Reply:
> Maybe this could be similar to what Alex's work showed about compendia where many studies were combined? Although you're seeing improved performance even for very small training sets (probably only a few studies), so that doesn't seem like it would explain the whole difference.

It could be? Though like you said I'm probably not working with enough studies for that to be the main factor.

I'm currently just using limma. A super abridged unsupervised comparison had them working roughly the same (https://github.com/greenelab/saged/blob/master/notebook/analysis/batch_effect_correction.ipynb). It's something to check out in the future, though my intuition is that different correction methods would lead to different magnitudes of the same issues.